script/mafia

```diff
@@ -26,16 +26,23 @@ run_upgrade () {
 
   trap clean_up EXIT
 
+  MAFIA_CUR="$0"
+
+  if [ -L "$MAFIA_CUR" ]; then
+    echo 'Refusing to overwrite a symlink; run `upgrade` from the canonical path.' >&2
```

Comment:
portable readlink is hard because the default "not-a-link" behaviour is different in all the implementations (the is a link behaviour is consistent). I have a shell 1-liner that I have used in the past to do it that works across *BSD, linux, solaris, but in attempting to remember it I tried to log on to my server at home and it appears my internode connection is out :grapes: